python/ray/serve/_private/application_state.py

```diff
@@ -141,27 +201,28 @@ def deploy(self, deployment_params: List[Dict]) -> List[str]:
                 "path in your application to avoid this issue."
             )
 
-        self._status = ApplicationStatus.DEPLOYING
-        return cur_deployments_to_delete
+        for params in deployment_params:
+            params["deployment_name"] = params.pop("name")
+            params["app_name"] = self._name
+
+        deployment_infos = {
+            params["deployment_name"]: deploy_args_to_deployment_info(**params)
+            for params in deployment_params
+        }
+        self._set_target_state(deployment_infos=deployment_infos)
 
     def update_obj_ref(self, deploy_obj_ref: ObjectRef, deployment_time: int):
+        """Update deploy object ref.
+
+        Resets the deployment timestamp, sets status to DEPLOYING, and wipes the current
+        target state. This means while the task to build the application is running, the
+        application state stops the reconciliation loop and waits on the task to finish
+        and call apply_deployments_args().
+        """
         self._deploy_obj_ref = deploy_obj_ref
```

Comment:
We should cancel the previous obj ref in the following scenario
- delete application.
- redeploy same application.